File: src/main/java/com/example/samla/funnel/FunnelManager.java
package com.example.samla.funnel;

import android.app.Activity;
import android.util.Log;
import com.example.samla.userinterface.ToastManager;

import java.util.ArrayList;

public class FunnelManager {
    private final static String TAG = FunnelManager.class.getSimpleName();

    private Activity activity;
    private ArrayList<WayPoint> wayPoints = new ArrayList<>();

    public FunnelManager (Activity activity) {
        this.activity = activity;
    }

    public void reachedWayPoint (int wayPoint) {
        wayPoints.add(new WayPoint(wayPoint));
        Log.i(TAG, "Waypoints: " + wayPoints.toString());
        ToastManager.makeToast(activity, "Reached Waypoint: " + wayPoint);
    }
}

File: src/main/java/com/samla/sdk/userflow/funnel/FunnelManager.kt
package com.samla.sdk.userflow.funnel

import android.util.Log
import com.samla.sdk.ISamla
import com.samla.sdk.userflow.Analytics.Companion.firebaseAnalytics
import com.samla.sdk.userinterface.ToastManager.makeToast
import java.util.*

object FunnelManager {
    private val TAG = FunnelManager::class.java.simpleName

    lateinit var ISamla: ISamla
    private val wayPoints = ArrayList<WayPoint>()

    fun setClientActivity(entry : ISamla) {
        ISamla = entry;
    }


    fun reachedWayPoint(wayPoint: Int) {
        wayPoints.add(WayPoint(wayPoint))
        Log.i(TAG, "Waypoints: $wayPoints")
        makeToast(ISamla.getActivity(), "Reached Waypoint: $wayPoint")
        firebaseAnalytics.setUserProperty(
            "milestone",
            wayPoint.toString()
        )
    }
}